### ISSUE TITLE
Keychain stopped working for me

### ISSUE THREAD

**Issue by wehmoen** [MattyIce/steem-keychain#82]
For a couple of days now I can not use keychain anymore. Everytime when a website requests me to approve a transaction this is all I see:
![image](https://user-images.githubusercontent.com/6608103/54057817-419acb00-41f4-11e9-8fae-0fc3400751da.png)
Neither the extensions nor the websites console show any errors. After many retries it works from time to time. Any ideas what could be wrong? I installed the extension from the chrome webstore. extension and browser are all on the latest version.

**Comment by stoodkev** [comment]
I ve tried to reproduce this on Linux and Windows, and on different browsers ( Chrome Opera Firefox) and I can t reproduce the bug.
Have you checked the log from the background ?

**Comment by inertia186** [comment]
I have also observed this behavior.  Seems like it happens only once in 24 hours.  I believe it has something to do with the DOM not being fully loaded.  On my own app, I can work around this by chaining my intended request inside a call to `requestHandshake`, which seems to allow the extension to load before my intended request fires:

```javascript
    function prepareCustomJson ( ) {
      if ( !!window.steem_keychain ) {
        steem_keychain.requestHandshake(requestCustomJson);
      } else {
        alert("Steem Keychain not detected.");
      }
    }
    
    function requestCustomJson ( ) {
      steem_keychain.requestCustomJson(account_name, custom_json_id, key_type, json, display_name, function(response) {
      	console.log(response);
        
        if ( response['success'] ) {
          window.location = 'dashboard';
        }
      });
    }
    
    $(window).on('load', prepareCustomJson);
```

**Comment by inertia186** [comment]
Now it happens 90% of the time.

**Comment by wehmoen** [comment]
it became worse over the last couple of days

**Comment by stoodkev** [comment]
Do you guys still have this issue since last release? (5 days ago).

**Comment by wehmoen** [comment]
its inconsistent. sometimes it works fine. but especially when my browser is in fullscreen it doesnt work well (am on mac)

**Comment by inertia186** [comment]
No, I think one of the updates must have addressed this issue.  I have not seen this issue for a while.

**Comment by wehmoen** [comment]
Not the extension is just super laggy. WHen i type into a field i have to change the focus to see what I have actually entered.

**Comment by stoodkev** [comment]
@wehmoen do you still have issues with this?

**Comment by wehmoen** [comment]
yes. still extreme laggy. trying to film that today and drop u a link

**Comment by wehmoen** [comment]
https://www.youtube.com/watch?v=at-RxYuq2_g @stoodkev
